koap/src/main/kotlin/Decoder.kt

```diff
@@ -351,8 +429,25 @@ private fun Byte.toCode(): Code = when (val code = toInt()) {
     165 -> ProxyingNotSupported     // 5.05
 
     else -> {
-        val `class` = (code shr 5) and 0b111
-        val detail = code and 0b11111
-        Code.Raw(`class`, detail)
+        val `class` = (this shr 5) and 0b111
+        val detail = this and 0b11111
+        Message.Code.Raw(`class`, detail)
     }
 }
+
+/**
+ * Reads specified number of [bytes] from [ByteArrayReader] receiver to acquire a number.
+ *
+ * @param bytes (count) to read from [ByteArrayReader] to build number
+ * @return value of number
+ */
+internal fun ByteArrayReader.readNumberOfLength(
+    bytes: Int
+): Long = when (bytes) {
+    0 -> 0L
+    1 -> readUByte().toLong()
+    2 -> readUShort().toLong()
+    3 -> readUInt24().toLong()
+    4 -> readUInt()
+    else -> throw IllegalArgumentException("Unsupported number length of $bytes bytes")
```

Comment:
Should there also be an `8 -> readLong()`? Not necessarily that we use it, just seems odd to skip one of the available functions.

Reply:
The `8` (bytes) is absent here because there are no mentions of "64-bit" in the RFC in re: to encoding/decoding.
As far as I know, the larges supported unsigned integer size specified by the RFC is 32-bit; which is good since I opted out of using the experimental unsigned integers in Kotlin (we'd have a lot of potential overflows to deal with if the RFC specified 64-bit **unsigned** integers.

Reply:
I overlooked that token is being read from 8 bytes (albeit as a **signed** integer).
@cedrickcooke you were correct, you tried to prevent a regression but I didn't listen. Oops
Fixed in #20.